3p/integration.js

```diff
@@ -375,21 +387,22 @@ const defaultAllowedTypesInCustomFrame = [
 
 
 /**
- * Gets data encoded in iframe name attribute.
- * @return {!JsonObject}
+ * Initialize 3p frame.
+ * @param {!Window} win
  */
-function getData(iframeName) {
-  try {
-    // TODO(bradfrizzell@): Change the data structure of the attributes
-    //    to make it less terrible.
-    return parseJson(iframeName)['attributes'];
-  } catch (err) {
-    if (!getMode().test) {
-      dev().info(
-          'INTEGRATION', 'Could not parse context from:', iframeName);
-    }
-    return FALLBACK_CONTEXT_DATA;
-  }
+function init(win) {
+  const data = getData();
```

Comment:
I was thinking of an eventual structure like this

```
config (the whole thing parsed from name):
- context
  - data
  - other APIs
- mode
  - experiment toggles
  - others
```

We can parse name once and put it as module level variable. There is no problem to have module level variable. It caused a problem simply because it has a bad name `data`, which was easily mis-used by other code.